ItemDescription3/tableEXT/salesLineTabExt.50401.TabExt.al

```diff
@@ -0,0 +1,26 @@
+tableextension 50401 "BCP salesLineTabExt" extends "Sales Line"
+{
+    fields
+    {
+        modify("No.")
+        {
+            trigger OnAfterValidate()
+            var
+                item: Record Item;
+            begin
+                if Type = Type::Item then begin
+                    item.Get("No.");
```

Comment:
What will be if we validate No. to empty value?

Reply:
where is the best position to check for empty value in "No." field? is it before if statement or after and is there any significant difference between placing it before or after?
i personally think it can come after the if statement, because at that point we are ready to get the value in the "No." field , so it make sense to check if it is empty at that point before getting it. correct me if i'm wrong?

Thanks

i

Reply:
For purchase & sales lines you don't have to check anything, because when someone validate no. to clear value standard call Rec.Init() procedure which will clear all values except of primary key
but your testfield or get will throw an error in this process, so your code is break standard

Best way it's to check No. before everything, because you don't need to do anything in this case.
If Rec."No." ='' then
  exit;
  
This early exit will skip your code at the beginning. Early exit is very good practice to exit from code if you know conditions where code should not be called.